modules/standard/GMPHelper/chplgmp.h

```diff
@@ -17,15 +17,10 @@
  * limitations under the License.
  */
 
-#ifdef CHPL_HAS_GMP
```

Comment:
Seems a little unusual to have C header files be a part of the chapel standard library. 

Looking at this file, I don't see any reason this needs to be in C, could these just be chapel functions that call extern procs?

Reply:
First, I agree. Second, this PR was intended to take the easiest approach, at least initially.

It's certainly the case that many if not all of these could be re-implemented in Chapel, but I didn't view taking the time to do so as required to make this change. If you think it *is* required, then I can find time to do that. I'm not yet sure if all of them can be implemented in Chapel (some of them are workarounds for LLVM issues with GMP headers).

Reply:
Personally, I'd be in favor of trying to convert it to a Chapel module, and if that takes more than an hour or two making it into an issue. 

I worry that if we put it off it'll basically never get prioritized because things work "well enough"

Reply:
@ronawho - I had another look & updated this PR to move as much as I can with reasonable effort to module code. It took more effort than expected to do so because in testing the GMPRandom copy-across-locales code it became clear that it wasn't working properly.

Small functions continue to exist in the C header for one reason or another:
 * Chapel module doesn't know how to look inside __mpz_struct / __gmp_randstate_struct
 * chpl_gmp_mpz_get_str: extern interoperability with strings is awkward
 * chpl_mpz_cmp_si ... -- the LLVM backend can't call these GMP functions because they are written as macros